lib-rpc/dune

```diff
@@ -1,4 +1,11 @@
+(library
+ (name ocamlformat_rpc)
+ (public_name ocamlformat-rpc)
+ (libraries ocamlformat ocamlformat-rpc-lib)
+ (modules ocamlformat_rpc))
```

Comment:
I also support having 1 library per directory, but to avoid moving the `bin/` files in this PR (and I like having the binaries in `bin/`), maybe it would be enough to move this new library into `lib-rpc-server/` or something similar?